at-angular.ts

```diff
@@ -107,8 +144,43 @@ module at {
             if (target.$inject && target.$inject.length > 0) {
                 factory.$inject = target.$inject.slice(0);
             }
-            angular.module(moduleName).factory(className, factory);
+            angular.module(moduleName).factory(className || getFuncName(target), factory);
+        };
+    }
+
+    export interface IDecoratorAnnotation {
+        (moduleName: string, targetProvider: string): IClassAnnotationDecorator;
+    }
+
+    export function decorator(moduleName: string, targetProvider: string): at.IClassAnnotationDecorator {
+
+        return (targetClass: any): void => {
+
+            angular
+                .module(moduleName)
+                .config([
+                    '$provide',
+                    function($provide: angular.auto.IProvideService): void {
+
+                        delegation.$inject = ['$delegate', '$injector'];
+
+                        function delegation(
+                            $delegate: angular.ISCEDelegateProvider,
+                            $injector: angular.auto.IInjectorService
+                        ): any {
+
+                            let instance: any = $injector.instantiate(targetClass, {
```

Comment:
This variable is used only once, so it can be removed - just place right hand part of expresion as 3rd param of extend below.